youtube_dl/extractor/ccma.py

```diff
@@ -13,7 +13,7 @@
 
 
 class CCMAIE(InfoExtractor):
-    _VALID_URL = r'https?://(?:www\.)?ccma\.cat/(?:[^/]+/)*?(?P<type>video|audio)/(?P<id>\d+)'
+    _VALID_URL = r'https?://(?:www\.)?ccma\.cat/(?P<channel>tv3|catradio|[^/*])(?:[^/]+/)*?(?P<type>audio|video|[^/]*)/(?P<id>\d+)/?$'
```

Comment:
`(?P<channel>tv3|catradio|[^/*])` no, this imposes more strictness.
`[^/*]` incorrect.
`[^/]*` matching empty string does not make sense.
`/?$` incorrect.